numpy/lib/npyio.py

```diff
@@ -715,13 +716,20 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
 
     """
     # Type conversions for Py3 convenience
-    comments = asbytes(comments)
+    if isinstance(comments, (np.compat.basestring, np.compat.bytes)):
+        comments = [asbytes(comments)]
+    else:
+        comments = [asbytes(comment) for comment in comments]
     user_converters = converters
     if delimiter is not None:
         delimiter = asbytes(delimiter)
     if usecols is not None:
         usecols = list(usecols)
 
+    # Compile regex for comments beforehand
+    regex_comments = re.compile(asbytes('|').join(re.escape(comment)
+                                                  for comment in comments))
```

Comment:
Not as easy to read as it could be. Might be better to separate the regex string definition so that it fits on one line.